transactions/src/main/java/org/jboss/as/txn/service/ArjunaRecoveryManagerService.java

```diff
@@ -155,16 +158,18 @@ public synchronized void start(StartContext context) throws StartException {
             }
         }
 
+        startCompensationsLocalRecovery();
```

Comment:
@ochaloup  If compensations is an optional module then starting the recovery module also needs to be optional (ie this needs to be configured externally).

Reply:
I am not sure about what is optional in this case?

However, I don't understand why the local recovery module is hidden in the operation rather than just following straight after the TXOJ module: https://github.com/ochaloup/wildfly/blob/62ed88fd4087663d2cde92c81cda96775eb6a783/transactions/src/main/java/org/jboss/as/txn/service/ArjunaRecoveryManagerService.java#L108

IIUC your concern about option, TXOJ is similarly "optional" - i.e. customer might not have used it but that is not guarded.

Reply:
@mmusgrov about the optional, I agree with Tom as it was my understanding - the recovery module is available, it can handle unfinished transaction in the log store (e.g. left from a some prior container startup) and it depends if customer has a application directly using it

@tomjenkinson true, I think the compensation local recovery can be handled as other ones. I was still have XTS dependency clash in my mind and didn't realize that it could be move to `ArjunaRecoveryManagerService` even I was considering it at first. I'm going to give a try to reimplement it as you suggest.

Reply:
XTS if an optional subsystem but this code is enabling XTS recovery regardless of whether XTS is loaded.

Reply:
@ochaloup you say "it can handle unfinished transaction in the log store left from a some prior container startup". If the XTS subsystem is  not enabled can we really recovery an XTS transaction!

@tomjenkinson what was your point about TXOJ being similarly "optional". Are you saying that we need to fix that too or are you saying there is a precedent for including recovery for optional subsystems?

Reply:
@mmusgrov here could be the misunderstanding possibly - the transaction subsystem enables only local compensation recovery.
The XTS subsystem enables remote (XTS based) compensation recovery. Thus when XTS is not enabled then XTS recovery is enabled neither.

And as Tom mentioned, and I agree with him, the local compensation recovery module should be loaded (if possible) in the same manner as other recovery modules configured under the transaction subsystem.

Reply:
@mmusgrov - the latter

Reply:
Ok thanks for the clarification. In that case I approve this PR.